### PR TITLE
dialects: LLVM array type

### PR DESCRIPTION
**Pull request by mesham**
The LLVM array type and tests for llvm.array

### CODE REVIEW

File: xdsl/dialects/llvm.py [hunk 1]
```diff
@@ -105,6 +105,43 @@ def is_typed(self):
         return not isinstance(self.type, NoneAttr)
 
 
+@irdl_attr_definition
+class LLVMArrayType(ParametrizedAttribute, MLIRType):
+    name = "llvm.array"
+
+    type: ParameterDef[Attribute]
+    size: ParameterDef[IntAttr]
```
Comment (webmiche) [inline xdsl/dialects/llvm.py:113]: I would reorder them to stay consistent with the printed IR. Not a big issue though :)

**Review by webmiche — APPROVED** [review]

File: xdsl/dialects/llvm.py [hunk 2]
```diff
@@ -105,6 +105,43 @@ def is_typed(self):
...
+            parser.parse_characters('>', "End of llvm.array type expected!")
+            return [None, size]
+        parser.parse_characters(
+            'x', "llvm.array size and type must be separated by `x`")
+        type = parser.try_parse_type()
+        if type is None:
+            parser.raise_error(
+                "Expected second parameter of llvm.array to be a type!")
+        parser.parse_characters('>', "End of llvm.array parameters expected!")
+        return [type, size]
```
[the remark this answers is not in the document]
Comment (AntonLydike) [reply xdsl/dialects/llvm.py:138]: Yeah, just ignore this

File: xdsl/dialects/llvm.py [hunk 3]
```diff
@@ -105,6 +105,43 @@ def is_typed(self):
...
+            'x', "llvm.array size and type must be separated by `x`")
+        type = parser.try_parse_type()
+        if type is None:
+            parser.raise_error(
+                "Expected second parameter of llvm.array to be a type!")
+        parser.parse_characters('>', "End of llvm.array parameters expected!")
+        return [size, type]
+
+    @staticmethod
+    def from_size_and_type(size: IntAttr, type: Attribute):
```
Comment (AntonLydike) [inline xdsl/dialects/llvm.py:141]: If you want, you can make `size` here accept `int | IntAttr` to make using it easier (and in-line with the other `.from...` methods)

**Review by AntonLydike — APPROVED** [review]
LGTM, only minor comments. Maybe you could add a test that tests the `from_size_and_type` method?

File: xdsl/dialects/llvm.py (hunk 1 above)
Comment (mesham) [reply xdsl/dialects/llvm.py:112]: Done :)

File: xdsl/dialects/llvm.py (hunk 3 above)
Comment (mesham) [reply xdsl/dialects/llvm.py]: Done :)

**Comment by mesham** [comment]
> LGTM, only minor comments. Maybe you could add a test that tests the `from_size_and_type` method?

Done :)

**Pull request merged**